sylvia/tests/reply.rs

```diff
@@ -1,39 +1,52 @@
 #![cfg(feature = "sv_replies")]
 
-use cosmwasm_std::SubMsgResult;
+use cosmwasm_std::{BankMsg, CosmosMsg, Empty, SubMsgResult};
 use cw_storage_plus::Item;
 use cw_utils::{parse_instantiate_response_data, ParseReplyError};
-use noop_contract::sv::Executor;
-use sv::{ALWAYS_REPLY_ID, FAILURE_REPLY_ID, REMOTE_INSTANTIATED_REPLY_ID, SUCCESS_REPLY_ID};
-use sylvia::cw_std::{to_json_binary, Addr, Binary, ReplyOn, Response, StdError, SubMsg, WasmMsg};
-use sylvia::types::{ExecCtx, InstantiateCtx, QueryCtx, Remote, ReplyCtx};
+use noop_contract::sv::{Executor, NoopContractInstantiateBuilder};
+use sv::{
+    SubMsgMethods, ALWAYS_REPLY_ID, FAILURE_REPLY_ID, REMOTE_INSTANTIATED_REPLY_ID,
+    SUCCESS_REPLY_ID,
+};
+use sylvia::builder::instantiate::InstantiateBuilder;
+use sylvia::cw_std::{Addr, Binary, Response, StdError, SubMsg};
+use sylvia::types::{CustomMsg, CustomQuery, ExecCtx, InstantiateCtx, QueryCtx, Remote, ReplyCtx};
 use sylvia::{contract, entry_points};
 use thiserror::Error;
 
 #[allow(dead_code)]
 mod noop_contract {
-    use cosmwasm_std::{StdError, StdResult};
-    use sylvia::types::{ExecCtx, InstantiateCtx};
+    use cosmwasm_std::{Empty, StdError, StdResult};
+    use sylvia::types::{CustomMsg, CustomQuery, ExecCtx, InstantiateCtx};
     use sylvia::{contract, entry_points};
 
     use sylvia::cw_std::Response;
 
-    pub struct NoopContract;
+    pub struct NoopContract<M, Q> {
+        _phantom: std::marker::PhantomData<(M, Q)>,
```

Comment:
Do we need to extend this contract to be generic? Tests often serves as examples, I think it would be good to maintain simple contracts testing basics, and then separate generic/more complex contract testing the more specific things.

Reply:
We could, however this would more or less lead to duplication of these tests and contracts.

Also we decided that using generic for custom message and custom query is a good practice.